src/Aspire.Hosting.Azure/Provisioning/Provisioners/AzureContainerAppsInfastructure.cs

```diff
@@ -0,0 +1,455 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Globalization;
+using System.Text;
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting.Azure;
+
+// Logic to generate compute and networking infrastructure for Azure Container Apps
+// based deployments.
+internal class AzureContainerAppsInfastructure(DistributedApplicationExecutionContext executionContext)
+{
+    public async Task GenerateAdditionalInfrastructureAsync(DistributedApplicationModel appModel, CancellationToken cancellationToken = default(CancellationToken))
+    {
+        bool isMode = executionContext.IsRunMode;
+        if (isMode)
+        {
+            return;
+        }
+
+        // Add the compute infrastructure
+
+        var containerAppEnv = new AzureBicepResource("containerAppEnv", templateString:
+            """
+            param location string
+            param tags object
+
+            var resourceToken = uniqueString(resourceGroup().id)
+
+            resource logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@2022-10-01' = {
+              name: 'law-${resourceToken}'
```

Comment:
For these "global" resources I wonder if we need some concept of the app name which we can then use when generating these resource names.